source/03_crs.txt

```diff
@@ -0,0 +1,110 @@
+published OGC Well Known Text standard with the following considerations:
+
+* The OGC Coordinate System WKT VLR data shall be a null-terminated string.
+* The OGC Coordinate System WKT VLR data shall be considered UTF-8.
+* The OGC Coordinate System WKT VLR data shall be considered C locale-based with
+  no localization of the numeric strings within the WKT being performed.
+
+
+.. note::
+
+  Any published version of an OGC WKT standard is a valid definition for LAS 1.5.
+  OGC has published multiple versions of WKT, including `18-010
+  <https://docs.ogc.org/is/18-010r11/18-010r11.pdf>`__ (commonly called "WKT2"
+  or "WKTv2") and `12-063r5 <https://docs.ogc.org/is/12-063r5/12-063r5.html>`__
+  (commonly called "WKT1" and the formally ratified variant of WKT that was used
+  for LAS 1.4). If OGC were to publish a new WKT standard, it would also be a
+  valid definition for LAS 1.5.
+
+.. note::
+
+    LAS 1.4 supported a dual definition of coordinate system depending on the
+    PDRF. PDRFs 0-5 were able to have their coordinate systems defined by
+    GeoTIFF keys, but they were removed in LAS 1.5. It is no longer possible 
+    or valid to define coordinate system information with GeoTIFF keys in LAS 1.5.
+
+.. _math_wkt:
+
+`OGC Math Transform WKT Record`_
+................................................................................
+
+**Optional**
+
+Writers may include an optional OGC Math Transform WKT Record to provide a
+supplemental parameterized math transform definition.
+
++-----------------+-----------------------------+
+| User ID         | LASF_Projection             |
++-----------------+-----------------------------+
+| Record ID       | 2111                        |
++-----------------+-----------------------------+
+
+This record must contain a Math Transform WKT consistent with the same WKT
+specification being used by :ref:`crs_wkt`, with the following considerations:
+
+* The OGC Math Transform WKT VLR data shall be a null-terminated string.
+* The OGC Math Transform WKT VLR data shall be considered UTF-8.
+* The OGC Math Transform WKT VLR data shall be considered C locale-based, and
+  no localization of the numeric strings within the WKT should be performed.
+
+.. _epoch:
+
+`Epoch Record`_
+................................................................................
+
+**Optional**
+
++-----------------+-----------------------------+
+| User ID         | LASF_Projection             |
++-----------------+-----------------------------+
+| Record ID       | 2113                        |
++-----------------+-----------------------------+
+
+The date against which coordinates in the LAS file are referenced in a dynamic
+coordinate reference system is expressed as a decimal year in the Gregorian
+calendar stored as a double. For example, an epoch of 2017-03-25 in the
+Gregorian calendar would be stored as ``2017.23`` in the VLR (``2017 + 84/365``).
```

Comment:
Add link to ISO(?) standard explaining how to calculate this?

Reply:
@rouault Sorry for the ping, but is there an ISO or IEEE standard for the floating point form of epochs?

Reply:
I don't know besides what ISO 19111 / OGC Abstract Topic 2 says:

```
3.1.27
epoch
<geodesy> point in time
Note 1 to entry: In this document an epoch is expressed in the Gregorian calendar as a decimal year.
EXAMPLE   2017-03-25 in the Gregorian calendar is epoch 2017.23.
```

and
```
B.4   Epoch
B.4.1    Introduction
In this document “Epoch” is a point in time. It is given as a decimal year in the Gregorian calendar, with yyyy.00 being midnight at the start of the 1st January of year “yyyy”. Should it be required to convert the epoch to or from a Gregorian calendar date, assuming any of 365, 365.25 or 366 days as the year length will be satisfactory for dealing with tectonic plate linear motion.
EXAMPLE:              2017-03-25 in the Gregorian calendar is epoch 2017.23.
```

Reply:
Thanks! I found this direct link to that document: https://docs.ogc.org/as/18-005r4/18-005r4.html#4, which we can reference.

Unfortunately, it seems incomplete to me. For example, is the denominator always 365, or does it become 366 for leap years? Is the numerator supposed to be the Julian Day (JD = 1 for Jan 1) or JD-1 (JD = 0 for Jan 1)?

Reply:
Ah, thank you I didn't scroll far enough to realize that the two citations differed from each other at https://docs.ogc.org/as/18-005r4/18-005r4.html#4 vs https://docs.ogc.org/as/18-005r4/18-005r4.html#65.

That should work to suit our needs, I think.